### ISSUE TITLE
mdTabs: ngRepeat and selected index

### ISSUE THREAD

**Issue by daniel-nagy** [angular/material#4030]
I have a service (a network of services really) that handles page navigation in my application. Essentially, I've implemented [two level navigation](https://www.google.com/design/spec/patterns/navigation.html#navigation-two-levels) using a left drawer and tabs.

At the top level I have what I call pages and each page may have multiple tabs. Each page keeps track of its active tab so when the user navigates between pages they are taken back to the last tab they had open. An example of my template is as follows,

```html
<md-tabs ng-if="page.tabs.length" md-selected="page.activeTab" md-center-tabs="false">
  <md-tab ng-repeat="tab in page.tabs | filter: tabFilter" label="{{tab.label}}" md-on-select="page.navigate()"></md-tab>
</md-tabs>
```

**The Problem**

The problem is, as tabs are removed the nearest safe index is calculated and because of the two-way data binding my activeTab is overwritten. So effectively, the active tab is always zero.

**Possible Solutions**

* Queue all tabs to be inserted or removed and only after all tabs are inserted or removed calculate the nearest safe index.

* Make the directive aware of `ng-repeat` and only after the `$last` item is inserted calculate the nearest safe index.

**Comment by yanndupuy** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  dupuyyann@gmail.com
    Domain zyapps.com has exceeded the max emails per hour (430/400 (107%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:40109 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by serveur.ex2-p10.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1avsZo-001Z9N-77
	for yann.dupuy@zyapps.com; Thu, 28 Apr 2016 22:30:22 +0200
Date: Thu, 28 Apr 2016 13:28:28 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1461875308;
	bh=MA+vY376M+zFMT7C2BXKUHUBGyBeByIwV9t4trYEz48=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=YFnnH1+m5gPFtTtZmUbpUNaSVXckTWzy5cjlk4BjANanN6u9m7eIp2dXunQoTqi0F
	 rnEl9rLR3CmXK9kXfmIzQDME2dvZxkoU1TiGJhe32FtQxk7lXCBQ8FUatKlux8rQAW
	 onnXvhbJavPkUkdTZb37Y61xocoNtWP59pTzVUas=
